cirq/work/observable_measurement_data.py

```diff
@@ -0,0 +1,478 @@
+            )
+
+    def variance(self, setting: InitObsSetting, *, atol: float = 1e-8):
+        """Compute the variance of the estimators of the given setting.
+
+        This is the normal variance divided by the number of samples to estimate
+        the certainty of our estimate of the mean. It is the standard error
+        of the mean, squared.
+
+        This uses `ddof=1` during the call to `np.var` for an unbiased estimator
+        of the variance in a hypothetical infinite population for consistency
+        with `BitstringAccumulator.covariance()` but differs from the default
+        for `np.var`.
+
+        Args:
+            setting: The setting
+            atol: The absolute tolerance for asserting coefficients are real.
+        """
+        if len(self.bitstrings) == 0:
+            # coverage: ignore
+            raise ValueError("No measurements")
+        self._validate_setting(setting, what='variance')
+
+        _, var = _stats_from_measurements(
+            bitstrings=self.bitstrings,
+            qubit_to_index=self._qubit_to_index,
+            observable=setting.observable,
+            atol=atol,
+        )
+
+        return var
+
+    def stderr(self, setting: InitObsSetting, *, atol: float = 1e-8):
+        """The standard error of the estimators for `setting`."""
+        return np.sqrt(self.variance(setting, atol=atol))
+
+    def means(self, *, atol: float = 1e-8) -> np.ndarray:
+        """Estimates of the means of the settings in this accumulator."""
+        return np.asarray([self.mean(setting, atol=atol) for setting in self.simul_settings])
+
+    def mean(self, setting: InitObsSetting, *, atol: float = 1e-8):
+        """Estimates of the mean of `setting`."""
+        if len(self.bitstrings) == 0:
+            # coverage: ignore
+            raise ValueError("No measurements")
+        self._validate_setting(setting, what='mean')
+
+        mean, _ = _stats_from_measurements(
+            bitstrings=self.bitstrings,
+            qubit_to_index=self._qubit_to_index,
+            observable=setting.observable,
+            atol=atol,
+        )
+
+        return mean
+
+
+def flatten_grouped_results(
+    grouped_results: List[BitstringAccumulator],
+) -> List[ObservableMeasuredResult]:
+    """Flatten results from a collection of BitstringAccumulators into a list
+    of ObservableMeasuredResult.
+
+    Raw results are contained in BitstringAccumulator which contains
+    structure related to how the observables were measured (i.e. their
+    grouping). This can be important for taking covariances into account.
+    This function removes that structure, giving a flat list of results
+    which may be easier to work with.
+
+    Args:
+        grouped_results: A list of BitstringAccumulators, probably returned
+            from `measure_observables` or `measure_grouped_settings`.
+    """
+    return list(itertools.chain.from_iterable(acc.results for acc in grouped_results))
```

Comment:
nit: a bit shorter and doesn't need itertools:
```suggestion
    return [res for acc in grouped_results for res in acc.results]
```